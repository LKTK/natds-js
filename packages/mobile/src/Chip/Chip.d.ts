declare const Chip: () => JSX.Element;
export default Chip;
