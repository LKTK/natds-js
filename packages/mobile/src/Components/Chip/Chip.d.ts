import PropTypes from 'prop-types';
declare const Chip: {
    ({ mode, children, icon, avatar, selected, selectedColor, disabled, accessibilityLabel, textStyle, style, theme, testID, onPress, onLongPress, onClose, ...other }: any): JSX.Element;
    propTypes: {
        mode: PropTypes.Requireable<string>;
        children: PropTypes.Requireable<PropTypes.ReactNodeLike>;
        icon: PropTypes.Requireable<any>;
        avatar: PropTypes.Requireable<PropTypes.ReactNodeLike>;
        selected: PropTypes.Requireable<boolean>;
        selectedColor: PropTypes.Requireable<string>;
        disabled: PropTypes.Requireable<boolean>;
        accessibilityLabel: PropTypes.Requireable<string>;
        textStyle: PropTypes.Requireable<any>;
        style: PropTypes.Requireable<any>;
        theme: PropTypes.Requireable<any>;
        testID: PropTypes.Requireable<string>;
        onPress: PropTypes.Requireable<(...args: any[]) => any>;
        onLongPress: PropTypes.Requireable<(...args: any[]) => any>;
        onClose: PropTypes.Requireable<(...args: any[]) => any>;
    };
    defaultProps: {
        mode: string;
        children: string;
        icon: string;
        avatar: string;
        selected: boolean;
        selectedColor: string;
        disabled: boolean;
        accessibilityLabel: string;
        textStyle: string;
        style: string;
        theme: string;
        testID: string;
        onPress: () => void;
        onLongPress: () => void;
        onClose: () => void;
    };
};
export default Chip;
