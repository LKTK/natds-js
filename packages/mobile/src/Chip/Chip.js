"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = __importDefault(require("react"));
var react_native_paper_1 = require("react-native-paper");
var Chip = function () { return react_1.default.createElement(react_native_paper_1.Chip, null, "Example Chip"); };
exports.default = Chip;
//# sourceMappingURL=Chip.js.map