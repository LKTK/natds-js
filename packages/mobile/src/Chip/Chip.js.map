{"version":3,"file":"Chip.js","sourceRoot":"","sources":["Chip.tsx"],"names":[],"mappings":";;;;;AAAA,gDAA0B;AAC1B,yDAAuD;AAEvD,IAAM,IAAI,GAAG,cAAM,OAAA,8BAAC,yBAAS,uBAAyB,EAAnC,CAAmC,CAAC;AACvD,kBAAe,IAAI,CAAC"}