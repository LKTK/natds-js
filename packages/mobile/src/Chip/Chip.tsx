import React from 'react';
import { Chip as PaperChip } from 'react-native-paper';

const Chip = () => <PaperChip>Example Chip</PaperChip>;
export default Chip;
