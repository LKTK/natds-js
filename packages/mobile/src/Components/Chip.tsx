import * as React from 'react';
import { Chip as PaperChip, withTheme } from 'react-native-paper';
import { StyleProp, ViewStyle, Surface, TextStyle } from 'react-native';
import { IconSource } from 'react-native-paper/lib/typescript/src/components/Icon';
import { IThemeShape } from 'Provider/IThemeShape';

declare type ChipProps = React.ComponentProps<typeof Surface> & {
  /**
   * Mode of the chip.
   * - `flat` - flat chip without outline.
   * - `outlined` - chip with an outline.
   */
  mode?: 'flat' | 'outlined';
  /**
   * Text content of the `Chip`.
   */
  children: React.ReactNode;
  /**
   * Icon to display for the `Chip`. Both icon and avatar cannot be specified.
   */
  icon?: IconSource;
  /**
   * Avatar to display for the `Chip`. Both icon and avatar cannot be specified.
   */
  avatar?: React.ReactNode;
  /**
   * Whether chip is selected.
   */
  selected?: boolean;
  /**
   * Whether to style the chip color as selected.
   */
  selectedColor?: string;
  /**
   * Whether the chip is disabled. A disabled chip is greyed out and `onPress` is not called on touch.
   */
  disabled?: boolean;
  /**
   * Accessibility label for the chip. This is read by the screen reader when the user taps the chip.
   */
  accessibilityLabel?: string;
  /**
   * Function to execute on press.
   */
  onPress?: () => void;
  /**
   * Function to execute on long press.
   */
  onLongPress?: () => void;
  /**
   * Function to execute on close button press. The close button appears only when this prop is specified.
   */
  onClose?: () => void;
  /**
   * Style of chip's text
   */
  textStyle?: TextStyle;
  /**
   * React Native style object
   */
  style?: StyleProp<ViewStyle>;
  /**
   * @optional
   */
  theme: IThemeShape;
  /**
   * Pass down testID from chip props to touchable for Detox tests.
   */
  testID?: string;
};

const Chip: React.FunctionComponent<Omit<ChipProps, 'height' | 'width'>> = (
  props: Omit<ChipProps, 'height' | 'width'>
) => <PaperChip {...props} />;

export default withTheme(Chip);
