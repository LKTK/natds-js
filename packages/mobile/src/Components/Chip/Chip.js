"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = __importDefault(require("react"));
var prop_types_1 = __importDefault(require("prop-types"));
var react_native_paper_1 = require("react-native-paper");
var Chip = function (_a) {
    var mode = _a.mode, children = _a.children, icon = _a.icon, avatar = _a.avatar, selected = _a.selected, selectedColor = _a.selectedColor, disabled = _a.disabled, accessibilityLabel = _a.accessibilityLabel, onPress = _a.onPress, onLongPress = _a.onLongPress, onClose = _a.onClose, textStyle = _a.textStyle, style = _a.style, theme = _a.theme, testID = _a.testID, other = __rest(_a, ["mode", "children", "icon", "avatar", "selected", "selectedColor", "disabled", "accessibilityLabel", "onPress", "onLongPress", "onClose", "textStyle", "style", "theme", "testID"]);
    return (react_1.default.createElement(react_native_paper_1.Chip, __assign({ mode: mode, selected: selected, icon: icon, avatar: avatar, selectedColor: selectedColor, disabled: disabled, accessibilityLabel: accessibilityLabel, textStyle: textStyle, style: style, theme: theme, testID: testID, onPress: onPress, onLongPress: onLongPress, onClose: onClose }, other), children));
};
Chip.propTypes = {
    mode: prop_types_1.default.string,
    children: prop_types_1.default.node,
    icon: prop_types_1.default.any,
    avatar: prop_types_1.default.node,
    selected: prop_types_1.default.bool,
    selectedColor: prop_types_1.default.string,
    disabled: prop_types_1.default.bool,
    accessibilityLabel: prop_types_1.default.string,
    textStyle: prop_types_1.default.any,
    style: prop_types_1.default.any,
    theme: prop_types_1.default.any,
    testID: prop_types_1.default.string,
    onPress: prop_types_1.default.func,
    onLongPress: prop_types_1.default.func,
    onClose: prop_types_1.default.func,
};
Chip.defaultProps = {
    mode: 'flat',
    children: 'Chip',
    icon: '',
    avatar: '',
    selected: false,
    selectedColor: '',
    disabled: false,
    accessibilityLabel: '',
    textStyle: '',
    style: '',
    theme: '',
    testID: '',
    onPress: function () { },
    onLongPress: function () { },
    onClose: function () { },
};
exports.default = Chip;
//# sourceMappingURL=Chip.js.map