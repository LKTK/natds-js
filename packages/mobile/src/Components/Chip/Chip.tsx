import { Chip as PaperChip } from 'react-native-paper';

export default class Chip extends PaperChip {}
