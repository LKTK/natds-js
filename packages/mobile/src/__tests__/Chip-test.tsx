import 'react-native';
import React from 'react';
import Chip from '../Chip/Chip';

import renderer from 'react-test-renderer';

it('renders chip correctly', () => {
  const chip = renderer.create(<Chip />).toJSON();
  expect(chip).toMatchSnapshot();
});
