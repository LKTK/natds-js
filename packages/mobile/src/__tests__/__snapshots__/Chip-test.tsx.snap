// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders chip correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ebebeb",
      "borderColor": "#ebebeb",
      "borderRadius": 16,
      "borderStyle": "solid",
      "borderWidth": 0.5,
      "elevation": 0,
    }
  }
>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    focusable={false}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 16,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "paddingHorizontal": 4,
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "Helvetica Neue",
              "textAlign": "left",
              "writingDirection": "ltr",
            },
            Array [
              Object {
                "lineHeight": 24,
                "marginVertical": 4,
                "minHeight": 24,
                "textAlignVertical": "center",
              },
              Object {
                "color": "rgba(0, 0, 0, 0.87)",
                "marginLeft": 8,
                "marginRight": 8,
              },
              undefined,
            ],
          ]
        }
      >
        Example Chip
      </Text>
    </View>
  </View>
</View>
`;
